cns/restserver/api.go

```diff
@@ -899,6 +899,95 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
 	logger.Response(service.Name, getNetworkContainerResponse, returnCode, err)
 }
 
+// getAllNetworkContainers is used to check if NC refresh association is needed
+// If any NC from DNC request is not existing in CNS, refresh association will be triggered
+func (service *HTTPRestService) getAllNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getAllNetworkContainers")
+	if r.Method != "GET" {
+		logger.Printf("[Azure CNS] Error. getAllNetworkContainers did not receive a GET.")
+		return
+	}
+
+	var req cns.GetAllNetworkContainersRequest
```

Comment:
- GET requests usually don't involve bodies. All the necessary information the server needs to handle the request should come from the URI.
- The purpose of this functionality is for DNC to learn about **any** NCs CNS is aware of. No need for this endpoint to take individual NC ids. If in the future that needs to be supported, they should be provided via query parameters on the URL.